### PR TITLE
[Cleanup] Adopt eslint vue/recommended: Grant Finder: search modals

### PR DESCRIPTION
**Pull request by sanason**
## Description
Brings Grant Finder search modals into compliance with new linting rules.

./src/components/Modals/AddKeyword.vue
./src/components/Modals/SavedSearchPanel.vue
./src/components/Modals/SearchPanel.vue
./src/components/SearchFilter.vue

All straightforward except:

- `AddKeyword` modal is not used anymore so it was deleted, not fixed.
- The auto-fixer broke edit and delete functionality in the `SavedSearchPanel`. See [comment](https://github.com/usdigitalresponse/usdr-gost/pull/2892/commits/6e250d15cb133f67df46753b0f7b32cfd8a6bd4f#r1546810917).

## Screenshots / Demo Video
No user-facing changes.

## Testing
Load up the app, go to the "Browse Grants" tab and check:

- SearchPanel: Click the "New Search" button, fill in search criteria, click "Save and View Results".
- SearchFilter:  When a search is active, check that the search criteria are listed at the top of the table. Click "Edit". Click "Clear".
- SavedSearchPanel: Click the "My Saved Searches" button. Test selecting an existing search. Test editing an existing search. Test deleting an existing search.

### Automated and Unit Tests
- [ ] Added Unit tests

### Manual tests for Reviewer
- [ ] Added steps to test feature/functionality manually

## Checklist
- [ ] Provided ticket and description
- [ ] Provided screenshots/demo
- [ ] Provided testing information
- [ ] Provided adequate test coverage for all new code
- [ ] Added PR reviewers

### CODE REVIEW

File: packages/client/src/components/Modals/SavedSearchPanel.vue [hunk 1]
```diff
@@ -73,14 +73,12 @@
                     />
                   </template>
                   <b-dropdown-item
-                    :search-id="data.item.id"
```
Comment (sanason) [inline packages/client/src/components/Modals/SavedSearchPanel.vue]: The auto-fixer actually broke this. It changed the attribute name from `:searchId=` to `:search-id=`, which I guess is fine if the attribute is a component prop, but which apparently doesn't work if the attribute is going to be looked up by name:

`const searchId = e.target.getAttribute('searchid');`

Comment (jeffsmohan) [reply packages/client/src/components/Modals/SavedSearchPanel.vue]: Good catch! 

I don't love anytime autofix breaks anything... but supplying a prop that the component doesn't expect just to store a value is pretty atypical usage. Thanks for the fix, that's much cleaner.

**Review by jeffsmohan — APPROVED** [review]
This looks great! Thanks for finding and fixing the couple of oddities here, with the prop data and the unused keyword functionality.

**Comment by jeffsmohan** [comment]
I'm going to go ahead and merge this for you, hope that's ok! I've got other in-flight PRs that might cause merge conflicts, so I'd rather get this landed so I can resolve those.

**Pull request merged**